server/src/internalClusterTest/java/org/opensearch/index/mapper/StarTreeMapperIT.java

```diff
@@ -275,7 +275,7 @@ public void testValidCompositeIndex() {
                     assertEquals(expectedMetrics, starTreeFieldType.getMetrics().get(0).getMetrics());
                     assertEquals(10000, starTreeFieldType.getStarTreeConfig().maxLeafDocs());
                     assertEquals(
-                        StarTreeFieldConfiguration.StarTreeBuildMode.ON_HEAP,
+                        StarTreeFieldConfiguration.StarTreeBuildMode.OFF_HEAP,
```

Comment:
let's randomize this

Reply:
This case is meant to assert the default build mode in star tree mapper, there is no scope for randomization here.

For now default is off heap.